Added portal powerup and optimized images

rahulraman0108 opened the pull request:
Hello, respected owners, collaborators, and contributors to this repository. I want to merge my few edits to this repository, I optimized the two images with the help of IMG bot and added a portal powerup.

Commit:
added portal powerup

ImgBotApp committed:
[ImgBot] Optimize images

*Total -- 3,030.88kb -> 2,682.13kb (11.51%)

/screenshots/SettingsWindow.png -- 833.41kb -> 666.12kb (20.07%)
/screenshots/ModManagerWindow.png -- 2,197.46kb -> 2,016.01kb (8.26%)

Commit:
Image optimizations

ImgBot optimized the two images in this repository.

Commit:
Image optimization

Optimized two images through IMG bot

Commit:
Changed the look of portal powerup

Changed the look of portal powerup to make it look better.

MattZ45986 commented:
@I-Am-The-Great, I like the idea, and as I was looking through the code, everything looks clean.  However, the random-position finder works in a way that portals can spawn where players fall through the map in probably undesirable ways (as demonstrated by poor Larry, as Zeus looks on in horror).  I would just modify some of the position-finding code to avoid this problem.  Other than that, it is a power-up that has the potential to make many mods very interesting.  Unless anyone else has issues, I would be willing to accept it once the modifications are made.

![portal](https://user-images.githubusercontent.com/33816924/50943694-76d8a600-145b-11e9-8013-a395c058fc23.JPG)

rahulraman0108 commented:
Hi, @MattZ45986 thanks for taking interest and participating in this PR. I am glad that you like my contribution, and the image is funny and it made me laugh at the situation :smile_cat: . Also in the image, I see the powerup shape is a sphere, which I modified in my latest commit `changed the look of the portal powerup`, so please see that look and say which one you like more.

MattZ45986 commented:
@I-Am-The-Great, I checked out the new look, and I do like the box shape better, but the texture might need some work.  Aaand... there is another problem (depending on how you look at it).  This time Larry was just minding his own business when all of a sudden he found himself stuck in a endless loop of falling through portals.  

Another suggestion: make the portals space out.  

Even if it's just to avoid this kind of situation, the portals should be some distance away from each other.  More practically, it allows for easy transportation across the map, which is what they were probably designed for in the first place.   It'll also give you a chance to put an original twist on @spdv123's code, which we always like to see. 
![portal2](https://user-images.githubusercontent.com/33816924/51091912-506a8180-175f-11e9-852a-a5af40533fa1.JPG)
This time, Zeus pretends to not notice.

rahulraman0108 commented:
Well, this image is not funny, hmmm I will work on the code, change the texture and make portals opening away from one another.

Commit:
Update Portal.py.

Updated the Portal.py to not let two portal openings open at the same position. Uhm I did not totally change the random position finder code by @spdv123 but modified it a little bit.

Commit:
Update snowyPowerup.py

Updated snowyPowerup.py to not let two portals open at the same position, added max portal number limits and changed the texture of the portal Powerup.

rahulraman0108 commented:
Hey, @MattZ45986 now check, if the problem is out, and if you like the new texture.

sudo-logic commented:
@I-Am-The-Great Impressive work with the Portals!
My server has had this powerup since 9 months. I had fixed the bugs with players falling off the maps etc.
Now all the portals open up inside the map bounds making sure that no one faces the problems that our little friend Larry is facing.
I still have one bug which I'm too lazy to fix, sometimes the portal opens up close to the host portal causing the endless loop.
It's not very disturbing as the portals only open up for 5 seconds after which Larry is able to move away peacefully.
@I-Am-The-Great If you face any problem, I'll release my portal scripts.

rahulraman0108 commented:
Hi @Awesome-Logic the first server which I joined in BS was yours only, and I liked your server very much an even now I like it, I got inspired to be editing or modding in BS only after seeing your servers and your skills, I wanted to develop my skills and enhance the BombSquad game, you were my inspiration for all the time. And this portal powerup's idea I got from your servers only, and I wanted it to be open source for BS players and fans so I made some scripts and opened this pull request, so that every player can have the fun of portals even at local level. In fact, every BS projects which I do is to enhance the fun in playing the BombSquad game.

sudo-logic commented:
Thanks a lot.😀

I'm thinking about making my server scripts open source for everyone.

It's been the most requested thing ever.

MattZ45986 commented on mods/Portal.py:
```diff
@@ -84,8 +106,13 @@ def getRandomPosition(self, activity):
                 pos[i][0] = min(pos[i][0], pt[i])
                 pos[i][1] = max(pos[i][1], pt[i])
-        # the credit of this random position finder goes to Deva
+        # The credit of this random position finder goes to Deva but I did some changes too.
```
Haha

MattZ45986 commented on mods/snowyPowerup.py:
```diff
@@ -179,8 +183,18 @@ def handleMessage(self,m):
+                                self.port = Portal.Portal(position1 = self.node.position,r = 0.9,color = (random.random(),random.random(),random.random()),activity = bs.getActivity())
+                                bs.gameTimer(t,bs.Call(self.delpor))
+                        else :
+                            bs.screenMessage('Only '+str(Portal.maxportals)+' pairs of portals are allowed at a time.')
```
We definitely don't need a reminder every time a portal powerup is collected past three.  I think people can figure it out by themselves.  Also, it won't happen too, too often, so it's better to just not interrupt game-flow in my opinion.

rahulraman0108 replied:
Hmm, k I remove now.

rahulraman0108 replied:
I did remove the screen message but the pull request yet shows change required, why is this so?

MattZ45986 commented on mods/snowyPowerup.py:
```diff
@@ -37,7 +37,7 @@ def __init__(self):
         self.powerdownSound = bs.getSound("powerdown01")
         self.dropSound = bs.getSound("boxDrop")
-        self.texPort = bs.getTexture("eyeColor")
+        self.texPort = bs.getTexture("ouyaOButton")
```
Much better! Good choice.

rahulraman0108 replied:
Thank you. :blush:

MattZ45986 commented:
Great work, @I-Am-The-Great.  I think portals will make a nice addition to the mod manager.  As soon as the screen message is removed, I consider the portal powerup to be @MattZ45986 approved.  If there are any other projects you would like to share, by all means do!

MattZ45986 reviewed (changes requested):
A visual reminder of the rules of the powerup are not necessary.  Instead of adding to the powerup, it probably distracts.  I do appreciate the thought, though.

Commit:
Update snowyPowerup.py

Removed visual notification of more than 3 portals being used.

rahulraman0108 commented:
![image](https://user-images.githubusercontent.com/42905362/51148460-50b06d00-1884-11e9-9ea6-04d026d7d3ae.png)
Well I did changes but why it yet say changes required? :thinking:

MattZ45986 commented:
I just had to approve your changes.  You are all set!  Very nice work.  I speak on behalf of the mod manager community when I say thank you for this fun (and might I add tried-and-tested) addition.

rahulraman0108 commented:
> 
> 
> I just had to approve your changes. You are all set! Very nice work. I speak on behalf of the mod manager community when I say thank you for this fun (and might I add tried-and-tested) addition.

Oh, the changes need to be approved, well this was my first Pull Request to any repository which is not my so I did not know it, well thank you for approving my changes and accepting my pull request, I am happy :smiley: .

And I want to open a pull request for a minigame, originally by Froshlee, but I added some modifications, it the bsSoccer game, it in I disabled bombing, punching and grabbing from players and added backflip on jump press, well I am making it currently today I will complete it, and please say if this idea is worthy to be added to this repository.

sritabh commented:
Isn't it like you're doing is copying others stuff and doing minor editing and taking whole credit for it?
Is it good or bad?

rahulraman0108 commented:
> 
> 
> Isn't it like you're doing is copying others stuff and doing minor editing and taking whole credit for it?
> Is it good or bad?

Lul :laughing: I am not taking credit of the snowyPowerup.py, and taking credit of making the portal powerup open source, in the Portal.py I have written that the original work of the random position finder is done by @spdv123 and I just did simple edits and about the bsSoccer game if approved then it will contain credits to Froshlee, and not to me, I would have just done simple edits in the script; so @SobyDamn your objection is totally wrong in my view.

I hope that you understand what I want to say.
